apps/roc_bindicator/roc_bindicator.star

```diff
@@ -0,0 +1,201 @@
+"""
+Applet: Roc Bindicator
+Summary: Show refuse in Rochester NY
+Description: Use address located in Rochester, NY to query the City website and then display what trash bins to but at the curb for collection.
+Author: Nolan Lynch
+"""
+load("render.star", "render")
+load("http.star", "http")
+load("schema.star", "schema")
+load("encoding/json.star", "json")
+
+DEFAULT_LOCATION = """
+{
+	"lat": "43.1529219",
+	"lng": "-77.6212444",
+	"description": "Rochester, NY, USA",
+	"locality": "Rochester",
+	"place_id": "ChIJCSF8lBZEwokRhngABHRcdoI",
+	"timezone": "America/New_York"
+}
+"""
+
+def main(config):
+
+#Get User Input Data from config
+    location = config.get("location", DEFAULT_LOCATION)
+    loc = json.decode(location)
+    Title_Texts = config.get("User_title_text", "Bindicator")
+    Bar_Color = config.get("bar_color", "#de00cb")
+    Positive_Color = config.get("Positive_Color", "#2f7000")
+    Negative_Color = config.get("negative_color", "#700000")
+
+#Convert config Location to Float from String
+    LatAsString = loc["lat"]
+    LatAsFloat = str(LatAsString)
+    LongAsString = loc["lng"]
+    LongAsFloat = str(LongAsString)
+
+#Set conversion URL. The Location needs to change from reference type 4326 to 2262 (This is what the City GIS system needs the Location data formatted in.)
+    Conversion_URL = "https://epsg.io/trans?x="+LongAsFloat+"&y="+LatAsFloat+"&s_srs=4326&t_srs=2262"
+
+#Get Contents of Conversion URL and if failure, tell us what the code is. Also set lat & long variables for use in next step
+    GIS = http.get(Conversion_URL)
```

Comment:
It would be good to add some caching arouns this (and other) HTTP calls so we keep request rates at a reasonable level.

Reply:
I don't know if I can add caching. Based on the documentation, isn't caching for the entire app installation, not a per-user cache? If we cached the location, it would be wrong for people not located at the same address.

Reply:
Correct, the cache is scoped to your app, but shared among users. You can construct your cache key so that it contains the location users in the call. That means only a user in the very same location will get the same data, which is ok.

Reply:
Great. Thank you for the clarification